cast/src/lib.rs

```diff
@@ -443,6 +445,62 @@ where
         Ok(self.provider.get_gas_price().await?)
     }
 
+    pub async fn ls(&self) -> Result<String> {
+        // create a vector of addresses
+        let mut addrs: Vec<H160> = Vec::new();
+
+        // fetch the chain id
+        let chain_id = self.chain_id().await?;
+
+        // for each file in $HOME/.ethereum/keystore
+        if let Ok(home_dir) = std::env::var("HOME") {
+             let path = std::path::Path::new(&home_dir).join(".ethereum").join("keystore");
+             let entries = fs::read_dir(path)?;
+             for entry in entries {
+                 let entry = entry?;
+                 let path = entry.path();
+                 let file = std::io::BufReader::new(std::fs::File::open(&path)?);
+                 let mut value: serde_json::Value = serde_json::from_reader(file)?;
+
+                 let map = value.as_object_mut().unwrap();
+                 let addr = map.get_mut("address").unwrap().as_str().unwrap();
+                 let t = H160::from_str(addr).unwrap();
```

Comment:
while it is (perhaps) unlikely that this fails, these unwraps here are a bit invasive.
alternatively, you can simply define a helper struct directly in this function like:

```rust
#[derive(Deserialize)
struct Entry {
   address: Address
} 
```